docs/source/data/subscriptions.mdx

```diff
@@ -366,159 +393,154 @@ subscription IncrementingNumber {
 
-###### `context`
+### `onConnect` and `onDisconnect`
 
-`ConnectionContext`
-</td>
-<td>
+Expanding on the above, you can add additional options for the subscription server to execute whenever a client connects (`onConnect`) or disconnects (`onDisconnect`).
 
-Context object for the WebSocket connection. This is _not_ the `context` object for the associated subscription operation.
+Defining an `onConnect` function lets you reject a particular incoming connection by returning false or by throwing an exception. This can be helpful if you'd like to check authentication when a client first connects to your subscription server.
 
-[See fields](https://github.com/apollographql/subscriptions-transport-ws/blob/master/src/server.ts#L35-L43)
+You provide these functions as options to the first argument of `useServer`, like so:
 
-</td>
-</tr>
+```ts{5,12}
+useServer(
+  {
+    schema,
+    // As before, ctx is the graphql-ws Context where connectionParams live.
+    onConnect: async (ctx) => {
+      // Check authentication every time a client connects.
+      if (tokenIsNotValid(ctx.connectionParams)) {
+        // You can return false to close the connection  or throw an explicit error
+        throw new Error('Auth token missing!');
+      }
+    },
+    onDisconnect(ctx, code, reason) {
+      console.log('Disconnected!');
+    },
+  },
+  wsServer,
+);
+```
+For more information and examples of using `onConnect` and `onDisconnect`, see the [`graphql-ws` documentation](https://github.com/enisdenjo/graphql-ws#ws-auth-handling).
 
-</tbody>
-</table>
+### Example: Authentication with the Server and Client
 
-### Example: Authentication with `onConnect`
+>If you plan to use subscriptions with Apollo Client, ensure **[both your client and server subscription protocols are consistent](https://www.apollographql.com/docs/react/data/subscriptions/#choice-of-subscription-protocol)** for the subscription library you are using (this example uses the `graphql-ws` library).
 
-On the client, `SubscriptionsClient` supports adding information to `connectionParams` ([example](https://www.apollographql.com/docs/react/data/subscriptions/#5-authenticate-over-websocket-optional)) that will be sent with the first WebSocket message. On the server, all GraphQL subscriptions are delayed until the connection has been fully authenticated and the `onConnect` callback returns a truthy value.
+On the client side, the `GraphQLWsLink` constructor supports adding information to `connectionParams` ([example](https://www.apollographql.com/docs/react/data/subscriptions/#5-authenticate-over-websocket-optional)). Those `connectionParams` are sent to your server when it connects, allowing you to extract information from the client request by accessing `Context.connectionParams`.
```

Comment:
"On the client side" -> "In Apollo Client"

Reply:
(We just want to be a little more Apollo-specific here because maybe someone's using our server but another library for client)